### PR TITLE
Class-wise saliency map generation for the detection task

### PR DESCRIPTION
**Pull request by negvet**
Enable class-wise saliency map generation for the detection case.

Tested with all detection models from `external\model-preparation-algorithm\configs\detection\` on PASCAL VOC dataset

related PR:
https://github.com/openvinotoolkit/training_extensions/pull/1402

Some examples (mobilenetv2_ssd):
image

### CODE REVIEW

File: mpa/det/inferrer.py [hunk 1]
```diff
@@ -159,30 +159,15 @@ def dump_features_hook(mod, inp, out):
...
-        saliency_map_hook = dump_saliency_hook if dump_saliency_map else dummy_dump_saliency_hook
 
         # Use a single gpu for testing. Set in both mm_val_dataloader and eval_model
         if is_module_wrapper(model):
             model = model.module
         with eval_model.module.backbone.register_forward_hook(feature_vector_hook):
-            with eval_model.module.backbone.register_forward_hook(saliency_map_hook):
+            with SaliencyMapHookDet(eval_model.module) if dump_saliency_map else nullcontext() as shook:
                 eval_predictions = single_gpu_test(eval_model, data_loader)
+                saliency_maps = shook.records if dump_saliency_map else [None] * len(self.dataset)
```
Comment (sungmanc) [inline mpa/det/inferrer.py]: How about using `saliency_hook` instead of `shook`?

File: mpa/modules/hooks/auxiliary_hooks.py [hunk 2]
```diff
@@ -132,6 +134,111 @@ def __exit__(self, exc_type, exc_value, traceback):
         self._handle.remove()
 
 
+class SaliencyMapHookDet(SaliencyMapHook):
```
Comment (sungmanc) [inline mpa/modules/hooks/auxiliary_hooks.py]: `DetSaliencyMapHook` is more attractable to me. Because this is also a kind of `Hook`.

**Review by goodsong81 — APPROVED** [review]
LGTM. I agree with @sungmanc for the naming, but not mandatory.

File: mpa/modules/hooks/auxiliary_hooks.py (hunk 2 above)
Comment (negvet) [reply mpa/modules/hooks/auxiliary_hooks.py]: I agree. Renamed

File: mpa/det/inferrer.py (hunk 1 above)
Comment (negvet) [reply mpa/det/inferrer.py]: Updated!

**Review by sungmanc — APPROVED** [review]
Thanks for reflecting comments, LGTM

**Pull request merged**